cloudsim-plus/src/main/java/org/cloudbus/cloudsim/hosts/HostSimple.java

```diff
@@ -525,6 +522,7 @@ public final Host setActive(final boolean activate) {
 
         this.active = activate;
         notifyStartupOrShutdown(activate, wasActive);
+        datacenter.schedule( 10, CloudSimTags.HOST_POWER_ON_OFF);
```

Comment:
You correctly defined a new tag and set the event, but the issue you are facing is because you didn't process that new event inside the target class DatacenterSimple (which is the base implementation for a Datacenter). Check the processEvent method.

Your Host.setActive method should have just this line you included, to request the Host activation (docs need to be updated).

The previous code from this method should be moved to a new method directly inside the HostSimple class.
This new method must be called inside the DatacenterSimple.processEvent I mentioned. 
Unfortunately, this new Host method must be public, otherwise, the Datacenter (which is in a different package) won't access it.

Reply:
**The previous code from this method should be moved to a new method directly inside the HostSimple class.**

By this, do you mean to say that the code except datacenter.schedule() call in Host.setActive() should be moved to a new method? Could you please tell me what portions of the code in  Host.setActive() you mean to transfer to a new function?

Reply:
Exactly, the entire previous code. The setActive should have just the code you'll introduced.
The code should be something such as:

```java
public final Host setActive(final boolean activate) {
    if(this.powerModel.getStartupDelay() > 0)
        datacenter.schedule( 10, CloudSimTags.HOST_POWER_ON_OFF);
    //if there is no delay, activate the host right away.
    else activation(activate);
} 

//Unfortunately, this has to be public or the Datacenter class won't access it.
public void activation(final boolean activate){
   //code previously on setActive
}
```